frontend/src/components/TopNavigation.tsx

```diff
@@ -44,17 +47,25 @@ interface Props {
  */
 const TopNavigation: FunctionComponent<Props> = ({ hideControls }) => {
   const { user, logout } = useAuth0()
+  const { profile } = useContext(UserProfileContext)
+
+  const filteredNavLinks = useMemo(() => {
```

Comment:
Suggestion (non-blocking): this isn't a strong opinion, but imo the additional code complexity and hit to readability isn't worth the performance improvement of memoizing things like this where N is likely to stay very small. In general I'd recommend only memoizing things that have a measurable performance improvement.